test/Concurrency/Runtime/async_taskgroup_discarding_noLeaks.swift

```diff
@@ -0,0 +1,119 @@
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library)
+
+// REQUIRES: concurrency
+// REQUIRES: executable_test
+// REQUIRES: concurrency_runtime
+
+// UNSUPPORTED: back_deployment_runtime
+// UNSUPPORTED: freestanding
+
+// FIXME: enable discarding taskgroup on windows; rdar://104762037
+// UNSUPPORTED: OS=windows-msvc
+
+import _Concurrency
+
+struct Boom: Error {
+  let id: String
+
+  init(file: String = #fileID, line: UInt = #line) {
+    self.id = "\(file):\(line)"
+  }
+
+  init(id: String) {
+    self.id = id
+  }
+}
+
+struct IgnoredBoom: Error {}
+
+@discardableResult
+func echo(_ i: Int) -> Int { i }
+@discardableResult
+func boom(file: String = #fileID, line: UInt = #line) throws -> Int { throw Boom(file: file, line: line) }
+
+func shouldEqual<T: Equatable>(_ lhs: T, _ rhs: T) {
+  precondition(lhs == rhs, "'\(lhs)' did not equal '\(rhs)'")
+}
+func shouldStartWith(_ lhs: Any, _ rhs: Any) {
+  let l = "\(lhs)"
+  let r = "\(rhs)"
+  precondition(l.prefix("\(r)".count) == r, "'\(l)' did not start with '\(r)'")
+}
+
+final class SomeClass: @unchecked Sendable {
```

Comment:
FIXME: this still will be leaked if it is a class; something is wrong with how we retain errors in `fillGroupNextErrorResult` 🤔 If we would not retain them there we crash though, so something is off and I don't quite get what yet.